server/controllers/dashboardController.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+import moment from "moment";
+import * as util from "util";
+import models from "../models";
+import { Op } from "sequelize";
+import paginationMeta from "../helpers/paginationHelper";
+import {paginationResponse} from "../utils/formatter";
+import {validFormat} from '../helpers/dateHelpers';
+
+
+const automation = models.Automation;
+/**
+ * Returns dateFunction
+ * @param {object} date - date string
+ * @returns {object} - date object
+ */
+const dateFunction = (date = { endDate: moment(), startDate: moment().subtract(30, 'days')}) => {  
+  // eslint-disable-next-line no-unused-vars2
+  const endDate = moment(date.endDate).format("YYYY-MM-DD");
+  const startDate = moment(date.startDate).format("YYYY-MM-DD");
+  // check if date.from is greater than date.to or today, return an error
+  if (startDate > endDate) {
+    throw new Error("startDate cannot be greater than endDate");
+  }
+  // check if both date from and to are provided
+  if (startDate && endDate) {
```

Comment:
Its expected that validations must have been done before passing date object as argument to the function, hence we may not need this extra check for: `if (startDate && endDate)`. Lets just return the result

Reply:
Is the expectation from the front-end?? If yes, I feel the backend may still need that validation

Reply:
I think its best we do the validation in a validator middleware instead of here